src/btwxt.cpp

```diff
@@ -11,32 +11,44 @@
 
 namespace Btwxt {
 
-RegularGridInterpolator::RegularGridInterpolator() = default;
+// RegularGridInterpolator::RegularGridInterpolator(GriddedData &grid_data_in)
+//     : grid_data(grid_data_in), grid_point(grid_data) {}
 
-RegularGridInterpolator::RegularGridInterpolator(GriddedData &grid_data_in)
-    : grid_data(grid_data_in), grid_point(grid_data) {}
+RegularGridInterpolator::RegularGridInterpolator(const std::vector<std::vector<double>> &grid)
+    : grid_data(grid), grid_point(grid_data) {}
```

Comment:
GriddedData and GridPoint classes are now relegated to be convenience objects only, and wholly compose RGI. 1. This makes the RGI interface much more similar to scipy.RegularGridInterpolator, and 2. Only one class, RGI, is exposed outside the library.